### DISCUSSION TITLE
Raspberry Pi 3/4 support (mpy)

### DISCUSSION THREAD

**Discussion by Tangerino** [micropython/micropython#9398] [General]
Dos mpy-cross supports RPi3/4 architectures?
Assuming MP was compiled and running in RPi, how do I compile the application sources py->mpy to run on these platforms?
Thank you

**Comment by Tangerino** [comment]
+info: may-cross also compiled in RPi

**Comment by scruss** [comment]
Not quite sure what you mean. mpy files should run as long as the bytecode version is compatible with the interpreter.

Small example on a Raspberry Pi Zero 2W (~ Pi 3, ish) using a freshly compiled MicroPython and mpy-cross:

*secrets.py*:

```python
secrets = {
    'ssid': 'thisone',
    'password': "nope",
    'hostname': "picow2",
}
```
Compile and delete the source, leaving only `secrets.mpy`:

    mpy-cross secrets.py && rm secrets.py

Now create *mpytest.py*:

```python
import secrets

print(secrets.secrets['hostname'])
```
Run it:

    micropython mpytest.py
    picow2

**Comment by Tangerino** [comment]
What I mean is:
I compiled MP and may-cross inside a RPI and when I try to run a single `print("A")` in the may it says :
`Invalid syntax` as in:
```bash
ubuntu@ubuntu:~$ cat x.py
print("it works!!!")

ubuntu@ubuntu:~$ mpy-cross x.py
ubuntu@ubuntu:~$ ls x.*
x.mpy  x.py
ubuntu@ubuntu:~$ ls -lh x.*
-rw-r--r-- 1 ubuntu ubuntu 42 Sep 25 10:58 x.mpy
-rw-rw-r-- 1 ubuntu ubuntu 22 Sep 24 11:10 x.py
ubuntu@ubuntu:~$ micropython x.py
it works!!!
ubuntu@ubuntu:~$ micropython x.mpy
Traceback (most recent call last):
  File "x.mpy", line 1
SyntaxError: invalid syntax
ubuntu@ubuntu:~$
```

Just to be sure, your example runs the `py` file, not the `mpy`

**Comment by scruss** [comment]
You can't run mpy files directly in any MicroPython interpreter

**Comment by jimmo** [comment]
@Tangerino  As a workaround, you can use

```bash
$ micropython -c "import x"
```

**Reply by Tangerino** [reply]
What I wish is to distribute the MPY files regardless the machine architecture (compiled accordingly) and not the source code.
